service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/NotifyGatekeeperController.java

```diff
@@ -51,18 +52,29 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToStartEvent(@RequestBody
         return respond(caseDetails, errors);
     }
 
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmitEvent(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseDetails caseDetailsBefore = callbackRequest.getCaseDetailsBefore();
+        CaseData caseData = getCaseData(caseDetails);
+        CaseData caseDataBefore = getCaseData(caseDetailsBefore);
+
+
+        if (SUBMITTED.equals(caseDataBefore.getState())) {
```

Comment:
what was wrong with the previous implementation? I know there was a bug when we used caseData.getState(), but why? about-to-submit hasn't completed yet, so shouldn't getState() return SUBMITTED? is it some async issue? @tomaszpowroznik

Reply:
@joez-f it was about ccd definition. It was changed for sendToGatekeeper to be 
PreConditionState: Submitted
PostConditionState: Gatekeeping.

in a callback you have case data as it will be when you save it, that why in controller state is Gatekeeping and not Submitted